router/transformer/transformer_test.go

```diff
@@ -0,0 +1,323 @@
+				Type:          "REST",
+				URL:           "http://www.ctx_timeout_dest.domain.com",
+				RequestMethod: http.MethodPost,
+				QueryParams:   map[string]interface{}{},
+				Body: map[string]interface{}{
+					"JSON": map[string]interface{}{
+						"key_1": "val_1",
+						"key_2": "val_2",
+					},
+					"FORM":       map[string]interface{}{},
+					"JSON_ARRAY": map[string]interface{}{},
+					"XML":        map[string]interface{}{},
+				},
+				Files: map[string]interface{}{},
+			},
+		},
+		"ctx_cancel_dest": {
+			name: "should throw timeout exception due to context getting cancelled immediately",
+			Expected: TcExpected{
+				StCode:      http.StatusInternalServerError,
+				Body:        `Post "%s/v0/destinations/ctx_cancel_dest/proxy": context canceled`,
+				ContentType: "text/plain; charset=utf-8",
+			},
+			Proxy: TcProxy{
+				StCode:   http.StatusOK,
+				Response: `{"output": {"status": 200, "message": "", "destinationResponse":"ctx_cancel_dest"}}`,
+			},
+			context: ProxyContextTc{
+				Cancel: true,
+			},
+			postParametersT: integrations.PostParametersT{
+				Type:          "REST",
+				URL:           "http://www.ctx_timeout_dest.domain.com",
+				RequestMethod: http.MethodPost,
+				QueryParams:   map[string]interface{}{},
+				Body: map[string]interface{}{
+					"JSON": map[string]interface{}{
+						"key_1": "val_1",
+						"key_2": "val_2",
+					},
+					"FORM":       map[string]interface{}{},
+					"JSON_ARRAY": map[string]interface{}{},
+					"XML":        map[string]interface{}{},
+				},
+				Files: map[string]interface{}{},
+			},
+		},
+		"not_found_dest": {
+			name: "should fail with not found error for not_found_dest",
+			Expected: TcExpected{
+				StCode:      http.StatusNotFound,
+				Body:        `post "%s/v0/destinations/not_found_dest/proxy" not found`,
+				ContentType: "text/plain; charset=utf-8",
+			},
+			Proxy: TcProxy{
+				StCode:   http.StatusNotFound,
+				Response: `Not Found`,
+			},
+			rtTimeout: 10 * time.Millisecond,
+			postParametersT: integrations.PostParametersT{
+				Type:          "REST",
+				URL:           "http://www.not_found_dest.domain.com",
+				RequestMethod: http.MethodPost,
+				QueryParams:   map[string]interface{}{},
+				Body: map[string]interface{}{
+					"JSON": map[string]interface{}{
+						"key_1": "val_1",
+						"key_2": "val_2",
+					},
+					"FORM":       map[string]interface{}{},
+					"JSON_ARRAY": map[string]interface{}{},
+					"XML":        map[string]interface{}{},
+				},
+				Files: map[string]interface{}{},
+			},
+		},
+	}
+)
+
+func Initialization() {
+	config.Load()
+	stats.Init()
+	stats.Setup()
+	rtTf.Init()
+	logger.Init()
+}
+
+func TestProxyRequest(t *testing.T) {
+	t.Setenv("RSERVER_HTTP_CLIENT_TIMEOUT", "1s")
+	Initialization()
+
+	srvMux := mux.NewRouter()
+	srvMux.HandleFunc("/v0/destinations/{destName}/proxy", proxyHandlerFunc)
+	srv := httptest.NewServer(srvMux)
+	defer srv.Close()
+
+	for destName, tc := range proxyResponseTcs {
```

Comment:
[minor] Using a map for test cases has the side effect of randomising test execution.

Which might create some confusion when debugging a test.

Reply:
Changed it